97 GitHub actions   automatic lint and test check

SamNie2027 opened the pull request:
### ℹ️ Issue

Closes #97

### 📝 Description

Added two Actions workflows: one to run jest and one to run the linter. They run on push and pr to main

Steps they take:

Jest:
Runs on Node 22.x, and 24.x (latest LTS major versions)
runs yarn install
creates .env based off of creating a .env with dummy values for aws. The AWS tests never actually try authing with AWS so they pass currently, the only test that requires real aws interaction is skipped.

Linter
Runs on Node 24.x
Yarn install
yarn lint

### ✔️ Verification

Latest pipeline passes. 

### Note
There are unfortunately a lot of tiny commits as in order to actually run the GitHub Action (see if it works) I have to commit.

SamNie2027 committed:
Removal of old CI/CD, adding jest and lint to run for two latest LTS node versions

SamNie2027 committed:
Fixing separation of name from run

SamNie2027 committed:
Define .env in the run and only run lint on latest node version (currently 24.x)

Define

SamNie2027 committed:
Update jest.yml

SamNie2027 committed:
Update jest.yml

SamNie2027 committed:
Update jest.yml

SamNie2027 committed:
Update jest.yml

SamNie2027 committed:
Update jest.yml

SamNie2027 committed:
Update jest.yml

SamNie2027 committed:
Update jest.yml

SamNie2027 committed:
Update jest.yml

SamNie2027 committed:
Removing this branch from ci

SamNie2027 committed:
Add build run

SamNie2027 committed:
Update build.yml

SamNie2027 committed:
Revert "Update build.yml"

This reverts commit 11de220e135c7a7bb3d42f61bbcc66699d485440.

SamNie2027 committed:
Revert "Add build run"

This reverts commit d38344726c79538a2bf3234ace970dd51e4adfb7.

SamNie2027 committed:
Merge branch 'main' into 97-github-actions---automatic-lint-and-test-check

rayyanmridha commented on .github/workflows/jest.yml:
```diff
@@ -0,0 +1,39 @@
+
+    strategy:
+      matrix:
+        node-version: [22.x, 24.x]
```
Is there a reason why we're also supporting 22.x? Shouldn't supporting the LTS like lint only work?

SamNie2027 committed:
Simplify Node.js version setup in Jest workflow

Removed matrix strategy for Node.js versions and fixed Node.js version to 24.x.

rayyanmridha reviewed (approved):
The new pipeline does what it's supposed to which seems good enough to me so we can close this one out.
